File: Exemplos/App_eXperience_Ionic/capacitor-elgin-m8/ios/Plugin/m8PluginPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(m8PluginPlugin, "m8Plugin",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
)

File: Exemplos/App_eXperience_Ionic/capacitor-elgin-m8/ios/Plugin/m8PluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(m8PluginPlugin)
public class m8PluginPlugin: CAPPlugin {
    private let implementation = m8Plugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
